WordPress/Sniffs/Functions/ReturnTypeSniff.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WordPress\Sniffs\Functions;
+
+use PHP_CodeSniffer_Tokens as Tokens;
+use WordPress\Sniff;
+
+/**
+ * Enforces formatting of return types.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.14.0
+ *
+ * @link    https://github.com/zendframework/zend-coding-standard/blob/52b435c714879609262aa36b004c4075cd967acc/src/ZendCodingStandard/Sniffs/Formatting/ReturnTypeSniff.php
+ */
+class ReturnTypeSniff extends Sniff {
+	/**
+	 * Simple return types.
+	 *
+	 * @since 0.14.0
+	 *
+	 * @var string[]
+	 */
+	private $simple_return_types = array(
+		'array'    => true,
+		'bool'     => true,
+		'callable' => true,
+		'double'   => true,
+		'float'    => true,
+		'int'      => true,
+		'iterable' => true,
+		'parent'   => true,
+		'self'     => true,
+		'string'   => true,
+		'void'     => true,
+	);
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return array(
+			T_RETURN_TYPE,
+		);
+	}
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @param int $stackPtr The position of the current token in the stack.
+	 * @return null Return before end if Return type declaration contains an invalid token.
+	 */
+	public function process_token( $stackPtr ) {
+		$colon = (int) $this->phpcsFile->findPrevious( T_COLON, $stackPtr - 1 );
```

Comment:
1. Don't cast the return to `(int)`, token index `0` exists.
2. You probably want to limit the search to the statement: `$this->phpcsFile->findPrevious( T_COLON, $stackPtr - 1, null, false, null, true );`
3. Always check that the return is not `false` before using it. Shouldn't ever happen in this case, but it's better to be safe than sorry. I've come across too many tokenizer bugs not to.
```php
if ( false === $colon ) {
	// Shouldn't happen, but just in case.
	return;
}